modules/core/src/adapters/deck-adapter.js

```diff
@@ -127,7 +127,8 @@ export default class DeckAdapter {
    * @param {(nextTimeMs: number) => void} proceedToNextFrame
    */
   onAfterRender(proceedToNextFrame) {
-    if (this.videoCapture.isRecording()) {
+    const areAllLayersLoaded = this.deck.props.layers.every(layer => layer.isLoaded);
```

Comment:
Very nice! By defining customRender you should be able to prevent rendering from happing until you are ready: https://github.com/visgl/deck.gl/blob/master/modules/core/src/lib/deck.js#L347

Reply:
That would mainly be an optimization to prevent wasted work....

Reply:
Would I call the private `this._drawLayers(redrawReason);` when I want to eventually render?

Reply:
exactly

Reply:
Made an issue to continue that work. Merging as-is for now since it's an improvement to the current renderer.